datastore/src/test/java/io/spine/server/storage/datastore/DsShardedWorkRegistryTest.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.storage.datastore;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.testing.NullPointerTester;
+import com.google.protobuf.Timestamp;
+import com.google.protobuf.util.Timestamps;
+import io.spine.base.Identifier;
+import io.spine.server.NodeId;
+import io.spine.server.delivery.ShardIndex;
+import io.spine.server.delivery.ShardProcessingSession;
+import io.spine.server.delivery.ShardSessionRecord;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.Optional;
+
+import static com.google.common.truth.Truth.assertThat;
+import static io.spine.server.storage.datastore.given.TestShardIndex.newIndex;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@DisplayName("DsShardedWorkRegistry should")
+public class DsShardedWorkRegistryTest {
+
+    private static final ShardIndex index = newIndex(1, 15);
+
+    private static final NodeId nodeId = newNode();
+
+    private final TestDatastoreStorageFactory factory =
+            TestDatastoreStorageFactory.defaultInstance();
+
+    private DsShardedWorkRegistry registry;
+
+    @BeforeEach
+    void setUp() {
+        factory.setUp();
+        registry = new DsShardedWorkRegistry(factory);
+    }
+
+    @AfterEach
+    void tearDown() {
+        factory.tearDown();
+    }
+
+    @Test
+    @DisplayName("pick up the shard and write a corresponding record to the storage")
+    public void pickUp() {
+        Optional<ShardProcessingSession> session = registry.pickUp(index, nodeId);
+        assertTrue(session.isPresent());
```

Comment:
Consider using `Truth8` for asserting the `Optional` value. It would be a bit shorter.

Reply:
I have considered that previously. It's not shorter, it's the same
```java
assertTrue(session.isPresent());    //  JUnit API.
assertThat(session).isPresent();    // Truth8 API.
```

And the following call checks the property of the instance wrapped by `Optional`. Unfortunately, there is no such an API call in Truth8:
```
// Impossible to check with `Truth8.assertThat(session).hasValue(..)`
assertThat(session.get()
                  .shardIndex()).isEqualTo(index);
```

And as long as the rest of the tests in this project aren't written in Truth, I have used Truth API only in those cases which would make the code more readable and the test failures easier to understand.